Coursera Google Advanced Data Analytics

Course 4 The Power of Statistics

Module 5 Hypothesis testing

Use Python to conduct a hypothesis test


In [ ]:
import pandas as pd
from scipy import stats

Use the relational operator for equals to get the relevant data from the state name column

Using the same random seed lets you generate the same set of random numbers in case you want to return to this sample data in the future.

In [ ]:
education_districtwise = pd.read_csv("/content/education_districtwise.csv")
education_districtwise = education_districtwise.dropna()
state21 = education_districtwise[education_districtwise['STATNAME'] == "STAT21"]
state28 = education_districtwise[education_districtwise['STATNAME'] == "STAT28"]
sampled_state21 = education_districtwise.sample(n=20, replace=True, random_state=13490)
sampled_state28 = education_districtwise.sample(n=20, replace=True, random_state=39103)

Use the mean function to compute the mean district literacy rate for both state 21 and state 28.

In [ ]:
sampled_state21['OVERALL_LI'].mean()

73.7335

In [ ]:
sampled_state28['OVERALL_LI'].mean()

68.832

The observed difference between the mean district literacy rates of state 21 and state 28 is below. At this point, state 21 has a higher overall literacy rate than state 28.

In [ ]:
sampled_state21['OVERALL_LI'].mean() - sampled_state28['OVERALL_LI'].mean()

4.901500000000013

Null hypothesis : There's no difference in the mean district literacy rates between State 21 and State 28.

Alternative hypothesis : There is a difference in the mean district literacy rates between State 21 and State 28.

`stats.ttest_ind(a,b,equal_var)`

*   `a`: observations from the first sample
*   `b`: observations from the second sample
*   `equal_var`: indicates whether the population variance of the two samples is assumed to be equal

Set equal_var to false, because you don't want to assume that the two samples have the same variance.

In [ ]:
stats.ttest_ind(a=sampled_state21['OVERALL_LI'], b=sampled_state28['OVERALL_LI'], equal_var=False)

TtestResult(statistic=1.6016356229508681, pvalue=0.11752736245983236, df=37.95608728152667)

**Draw a conclusion**

*   If p-value < significance level: **reject** the null hypothesis
    *   Statistially significant difference in literacy rates

*   If p-value > significance level: **fail to reject** the null hypothesis
    *   No statistially significant difference in literacy rates


P-value : 11.8% > Significance level : 5%

Fail to reject the null hypothesis and conclude that there is no a statistically significant difference between the mean district literacy rates of the two states, State 21 and State 28.

 You analysis will help the department decide how to distribute government resources. Since there is no a statistically significant difference in mean district literacy rates, the State 28 and the State 21, will likely receive same resources to improve literacy.